include/ensmallen_bits/utility/proxies.hpp

```diff
@@ -0,0 +1,294 @@
+/**
+ * @file proxies.hpp
+ * @author Marcus Edel
+ *
+ * Simple proxies that based on the data type forwards to `coot` or `arma`.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_UTILITY_PROXIES_HPP
+#define ENSMALLEN_UTILITY_PROXIES_HPP
+
+#include "function_traits.hpp"
+
+namespace ens {
+
+/**
+ * Helper struct that based on the data type `MatType` forwards to the
+ * corresponding `coot` or `arma` types. For example:
+ * If `MatType` is an `arma::mat`, then `ForwardType<MatType>::bmat` will be an
+ * `arma::Mat<ElemType>`.
+ * If `MatType` is a `coot::mat`, then `ForwardType<MatType>::bmat` will be a
+ * `coot::Mat<ElemType>`.
+ *
+ * This allows for writing generic code that can work with both `coot` and
+ * `arma` types without needing to know which library is being used at compile
+ * time.
+ */
+template<typename MatType, typename ElemType = typename MatType::elem_type>
+struct ForwardType
+{
+    // If MatType is an Bandicoot type, then we use Bandicoot types.
+    #ifdef USE_COOT
```

Comment:
I think it might be simpler here to use SFINAE to specialize the struct for all Bandicoot types at once, instead of using `std::conditional`, but I don't think it's a big deal either way---let me know what you think.  Happy to take a stab at it if you want.

Reply:
If you can take a look at this one I would appreciated it.

Reply:
Done in 44b215fa. :+1: